src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -289,8 +306,22 @@ private static void addBody(HttpURLConnection connection, Request<?> request, by
             connection.setRequestProperty(
                     HttpHeaderParser.HEADER_CONTENT_TYPE, request.getBodyContentType());
         }
-        DataOutputStream out = new DataOutputStream(connection.getOutputStream());
+        OutputStream out = this.createOutputStream ( request, connection, body.length );
         out.write(body);
-        out.close();
+        out.close ();
+    }
+
+    /**
+     * Overload this method for manipulate request stream (Use 'write (byte[] b)' in subclasse).
```

Comment:
Two comments here:

1) I think the wording could be clearer. In general a Javadoc should have a one-sentence summary of what the method needs to do, and then any additional context can come in other paragraphs. For example:

   "Create and return an OutputStream to which the request body will be written.

   <p>May be overridden by subclasses to manipulate or monitor this output stream."

2) My other comment is what the "Use write(byte[] b) in subclasse(s)" is referencing. What does this mean? In Volley itself, I'm assuming it's fine for us to use any valid method in OutputStream's signature to write data. We currently use write(byte[]) but I don't think we can guarantee always doing this in the future. But that's in Volley itself. In subclasses, why would you ever be calling write()?

   I *think* what you're saying here is that subclasses can override write(byte[]) to monitor the write. If so, I don't think we should document this, because that might change (maybe we buffer the writes and do it in chunks, for example). All we guarantee is that we'll write the bytes using OutputStream. Your subclass and subclasses in general should monitor any and all methods you care about (e.g. all three write overloads) and support any of them being called.